### PR TITLE
ARACHNE-2768 Datanode 1.15 does not use proxy settings in case of ARA…

### PR DESCRIPTION
**Pull request by YaroslavTir**
Arachne 2768 add proxy supprot for rest authentication

### CODE REVIEW

File: src/test/java/org/ohdsi/authenticator/service/rest/RestTemplateProviderTest.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.RestTemplate;
+
+public class RestTemplateProviderTest {
+
+    public static final String TEST_URL = "https://www.arachnenetwork.com/api/v1/build-number";
+    //this proxy ip is not fixed, so it could change. To fix this use any available proxy
+    public static final String PROXY_URL = "167.99.232.18";
```
Comment (pavgra) [inline src/test/java/org/ohdsi/authenticator/service/rest/RestTemplateProviderTest.java]: Where is this magic IP from? Cannot we avoid having new tests requiring an internet connection to complete?

**Review by pavgra — CHANGES_REQUESTED** [review]

File: src/test/java/org/ohdsi/authenticator/service/rest/RestTemplateProviderTest.java (hunk 1 above)
Comment (YaroslavTir) [reply src/test/java/org/ohdsi/authenticator/service/rest/RestTemplateProviderTest.java]: good point, I don't like this either. This is a random proxy that I have found on the internet. I plan to replace this one with our own proxy that Konstantin is working on. There is an option to set up an embedded proxy for all unit tests(timewise, I am not sure that it worth it ). and the last option is just to get rid of the test.

Comment (pavgra) [reply src/test/java/org/ohdsi/authenticator/service/rest/RestTemplateProviderTest.java]: > I plan to replace this one with our own proxy that Konstantin is working on

That doesn't solve the conceptual issue

> There is an option to set up an embedded proxy for all unit tests

Yes, that's what we should go for

Comment (YaroslavTir) [reply src/test/java/org/ohdsi/authenticator/service/rest/RestTemplateProviderTest.java]: fixed

File: src/test/java/org/ohdsi/authenticator/service/rest/RestTemplateProviderTest.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                        .host(PROXY_URL)
+                        .port(PROXY_PORT)
+                        .authEnabled(false)
+                        .build()
+        );
+
+        RestTemplate restTemplate = restTemplateProvider.createRestTemplate();
+        ResponseEntity<String> response = restTemplate.exchange(TEST_URL, HttpMethod.GET, null, String.class);
+
+        assertEquals(HttpStatus.OK, response.getStatusCode());
```
Comment (pavgra) [inline src/test/java/org/ohdsi/authenticator/service/rest/RestTemplateProviderTest.java:71]: I would add Mockito and check that proxy has actually been invoked

File: src/test/java/org/ohdsi/authenticator/service/proxy/ProxyInitializer.java [hunk 3]
```diff
@@ -0,0 +1,29 @@
+package org.ohdsi.authenticator.service.proxy;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.junit.rules.ExternalResource;
+import org.junit.rules.TestRule;
+
+
+/**
+ * Proxy should be start only ONCE for all unit tests!!
```
Comment (pavgra) [inline src/test/java/org/ohdsi/authenticator/service/proxy/ProxyInitializer.java]: Language

**Review by pavgra — CHANGES_REQUESTED** [review]

File: src/test/java/org/ohdsi/authenticator/service/rest/RestTemplateProviderTest.java (hunk 2 above)
Comment (YaroslavTir) [reply src/test/java/org/ohdsi/authenticator/service/rest/RestTemplateProviderTest.java:71]: fixed

File: src/test/java/org/ohdsi/authenticator/service/proxy/ProxyInitializer.java (hunk 3 above)
Comment (YaroslavTir) [reply src/test/java/org/ohdsi/authenticator/service/proxy/ProxyInitializer.java]: fixed

**Review by pavgra — APPROVED** [review]

**Pull request merged**